functions/src/lobby.ts

```diff
@@ -207,6 +219,56 @@ export const leaveLobby = functions.https.onCall((data: unknown, context): Promi
   });
 });
 
+export const applyLobbySettings = functions.https.onCall(async (data: unknown, context): Promise<void> => {
+  const auth = context.auth;
+  // no auth then you shouldn't be here
+  if (auth === undefined) {
+    throw new functions.https.HttpsError("permission-denied", "Not Signed In");
+  }
+  // validate code
+  if (!isLobbySettingsRequest(data)) {
+    throw new functions.https.HttpsError("invalid-argument", "Invalid lobby code!");
+  }
+  await db.runTransaction(async (transaction) => {
+    // get lobby doc
+    const lobby = lobbyCollection.doc(data.code);
+    const lobbyInfo = await transaction.get(lobby);
+
+    if (lobbyInfo.exists === false) {
+      throw new functions.https.HttpsError("not-found", "Lobby doesn't exist!");
+    }
+    // check if the request is coming from the host of the game
+    const { uids } = lobbyInfo.data() as Lobby;
+    if (auth.uid !== uids[0]) {
+      throw new functions.https.HttpsError("permission-denied", "Not the host of the game!");
+    }
+
+    if (data.lobbySettings.catfishAmount < 1 || data.lobbySettings.catfishAmount > 3) {
+      throw new functions.https.HttpsError("permission-denied", "Cannot have less than 1 or more than 3 catfish!");
+    }
+
+    if (
+      data.lobbySettings.promptTime < GAME_STATE_DURATIONS_MIN.PROMPT ||
+      data.lobbySettings.promptTime > GAME_STATE_DURATIONS_MAX.PROMPT ||
+      data.lobbySettings.chatTime < GAME_STATE_DURATIONS_MIN.CHAT ||
+      data.lobbySettings.chatTime > GAME_STATE_DURATIONS_MAX.CHAT ||
+      data.lobbySettings.voteTime < GAME_STATE_DURATIONS_MIN.VOTE ||
+      data.lobbySettings.voteTime > GAME_STATE_DURATIONS_MAX.VOTE
+    ) {
+      throw new functions.https.HttpsError("permission-denied", "Timer must be within the proper range!");
+    }
```

Comment:
these could be before the transaction so that if it fails then we don't need to do a database write if we get invalid data